apps/sharedtest/test/unit/wifi_helper_test.js

```diff
@@ -5,6 +5,122 @@ require('/shared/js/wifi_helper.js');
+                        'Input should be valid with 26 HEX char');
+
+      password = 'ABCGH12345ABCGH12345123456'; // 26 HEX invalid char
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+                        'Input should be invalid with 26 not HEX char');
+
+      password = 'ABCDEF123456ABCDEF123456ABCDEF12'; // 32 HEX char
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+                        'Input should be valid with 32 HEX char');
+
+      password = 'ABCGH12345ABCGH12345123456GGGG12'; // 32 HEX invalid char
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+                        'Input should be invalid with 32 not HEX char');
+
+      // 26 HEX char
+      password = 'ABCDEF123456ABCDEF123456ABABCDEF123456ABCDEF123456AB123456';
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+                        'Input should be valid with 58 HEX char');
+
+      // 58 HEX invalid char
+      password = 'ABCGH12345ABCGH12345123456ABCGH12345ABCGH12345123456ABCGH1';
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+                        'Input should be invalid with 58 not HEX char');
+    });
+
+    test('Test WPA-PSK network >', function() {
+      var network_security = 'WPA-PSK';
+      var password = '';
+
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+                              'Input should be invalid if not using password');
+
+      password = '12345';
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+                          'Input should be invalid if password is too short');
+
+      password = '1234567';
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+                          'Input should be invalid if password is too short');
+
+      password = '12345678';
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+                          'Input should be valid if password is long enough');
+
+      password = 'QWERTY123456';
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+                          'Input should be valid if password is long enough');
+    });
+
+    test('Test WPA-EAP network >', function() {
+      var network_security = 'WPA-EAP';
+      var password = '';
+      var user = '';
+
+      assert.isFalse(WifiHelper.isValidInput(network_security, 
+            password, user), 
+            'Input should be invalid when password and identity not filled ');
+
+      password = '123'; // password ok
+      assert.isFalse(WifiHelper.isValidInput(network_security, 
+          password, user), 'Input should be invalid when identity not filled');
+
+      user = 'fos'; // identity ok
+      assert.isTrue(WifiHelper.isValidInput(network_security,
+              password, user), 'Input should be valid when all conditions ok');
```

Comment:
I prefer to break these three assertions into three different test cases under a test suite. Because for this test suite, you can notice that each assertion will rely on used variable (password / user) and this would be hard to maintain if our rules are changed. For test cases, they should be more de-coupbled with each other.

And what you can do is add a `setup` block within this test suite to make sure in each test case, password & identity would be cleaned up.